### ISSUE TITLE
Make the `constructor` property of td.replaced() instantiable types useful

### ISSUE THREAD

**Issue by searls** [testdouble/testdouble.js#121]
Right now if you use `td.replace` on an instantiable function, what gets returned includes a `constructor` property, but it's not actually able to be used to stub/verify calls to `new Thing()` in the subject.

This was an oversight on my part because I hadn't thought of it, and I'm not quite sure what to do about it.

h/t @appleton via [twitter](https://twitter.com/appltn/status/756181200016814082)

**Comment by alexblom** [comment]
This would be really useful for me. In my case, I'm creating non-external objects with `new Object({params})`, and where passing the correct params is very important.  I'm unsure how to test this behaviour with testdouble today.

As an example, in a larger function you will see:
```
    validatePlugin = new ValidatePluginTask({
      project: this.project,
      ui: this.ui,
      platform: platform,
      pluginName: 'cordova-plugin-whitelist'
    });
```

I *really* care that nobody forgot to pass pluginName, and that value of pluginName matches expectations.

**Comment by searls** [comment]
Right on @alexblom -- I think this is important as well, and you have a very valid point. Unfortunately, I'm slammed with talk prep and travel the next month or two and don't want to commit to doing this myself soon.

**Comment by searls** [comment]
This will be fixed as an acceptance criteria of #166, so I'm closing it here
